schema.cql

```diff
@@ -0,0 +1,56 @@
+CREATE KEYSPACE IF NOT EXISTS sensors WITH replication = {'class': 'SimpleStrategy', 'replication_factor' : 1} ;
+USE sensors;
+CREATE TABLE IF NOT EXISTS sensors (
+	ip INET
+, name TEXT
+,	last_read TIMESTAMP
+,	online BOOLEAN
+,	PRIMARY KEY(ip)
+);
+CREATE TABLE IF NOT EXISTS sensors_data (
+	ip INET
+, name TEXT
+, date DATE
+, datetime TIMESTAMP
+, temperature FLOAT
+, relative_humidity FLOAT
+, dew_point FLOAT
+, co2_level FLOAT
+, PRIMARY KEY(ip, datetime)
```

Comment:
- To avoid unbounded partions, I believe it's better to set the primary key to: 
PRIMARY KEY ((ip, date), datetime)
- Set index for date:
CREATE INDEX info_date ON sensors.sensors_data (date);

Reply:
@ahmad-alsabbagh-clf after adding these changes, I get an error on starting sensor_data.py:

`Exception in thread stats_thread:
Traceback (most recent call last):
  File "/usr/lib64/python3.6/threading.py", line 916, in _bootstrap_inner
    self.run()
  File "/usr/lib64/python3.6/threading.py", line 864, in run
    self._target(*self._args, **self._kwargs)
  File "/data/comet-web-sensor/stats_writer.py", line 55, in start
    stats_data = self._process_stats_data()
  File "/data/comet-web-sensor/stats_writer.py", line 34, in _process_stats_data
    df = self._filter_dataframe_time_window(df)
  File "/data/comet-web-sensor/stats_writer.py", line 39, in _filter_dataframe_time_window
    df = df.set_index("datetime").between_time(self._work_day_start, self._work_day_end).reset_index()
  File "/data/comet-web-sensor/venv/lib/python3.6/site-packages/pandas/core/generic.py", line 7685, in between_time
    raise TypeError("Index must be DatetimeIndex")
TypeError: Index must be DatetimeIndex`

Reply:
Thanks,
That's because the sensor_data table is empty and it's trying to filter the data on an empty frame and it has nothing to do with previous changes. However, if you leave the sensor_data.py running for a couple of minutes then it would've stored already some data in the table and you won't get the error if you stopped it and run it once again.

The current development branch has a newer version of DAO class and I'll be rectifying this issue alongside merging your changes back into development today